src/server.ts

```diff
@@ -46,3 +50,143 @@ async function connectToDBAndStartListening() {
+    try {
+        const { front, back, deck_id } = req.body;
+        const text =
+            "INSERT INTO flashcard(front, back, deck_id) VALUES($1, $2, $3)";
+        const values = [front, back, deck_id];
+        qNum++;
+        const response = await queryAndLog(qNum, client, text, values);
+        res.status(200).json(response);
+    } catch (err) {
+        console.error(err);
+    }
+});
+
+app.post("/decks", async (req, res) => {
+    try {
+        const { name, user_id } = req.body;
+        const text = "INSERT INTO deck(name, user_id) VALUES($1, $2)";
+        const values = [name, user_id];
+        qNum++;
+        const response = await queryAndLog(qNum, client, text, values);
+        res.status(200).json(response);
+    } catch (err) {
+        console.error(err);
+    }
+});
+
+app.post("/users", async (req, res) => {
+    try {
+        const { user_name } = req.body;
+        const text = "INSERT INTO users(name) VALUES($1)";
+        const values = [user_name];
+        qNum++;
+        const response = await queryAndLog(qNum, client, text, values);
+        res.status(200).json(response);
+    } catch (err) {
+        console.error(err);
+    }
+});
+
+app.get("/users", async (_req, res) => {
+    try {
+        const text = "SELECT * FROM users";
+        qNum++;
+        const response = await queryAndLog(qNum, client, text);
+        const allUsers = response.rows;
+        res.status(200).json(allUsers);
+    } catch (err) {
+        console.error(err);
+    }
+});
+
+app.put("/flashcards/:id", async (req, res) => {
+    try {
+        const { streak } = req.body;
+        const { id } = req.params;
+        const interval = getInterval(streak);
+        const hours = `${interval * 24} hours`;
+        const text =
+            "UPDATE flashcard SET next_review = now() + $1, streak = $2 WHERE card_id = $3";
+        const values = [hours, streak, id];
+        const response = await queryAndLog(qNum, client, text, values);
+        res.status(200).json(response);
+    } catch (err) {
+        console.error(err);
+    }
+});
+
+app.delete("/flashcards/:id", async (req, res) => {
+    try {
+        const { id } = req.params;
+        const text = "DELETE from flashcard WHERE card_id = $1";
+        const values = [id];
+        const response = await queryAndLog(qNum, client, text, values);
+        res.status(200).json(response);
+    } catch (err) {
+        console.error(err);
+    }
+});
+
+app.delete("/decks/:id", async (req, res) => {
+    try {
+        const { id } = req.params;
+        const cardsText = "DELETE from flashcard WHERE deck_id = $1";
+        const values = [id];
+        const decksText = "DELETE from deck WHERE deck_id = $1";
+        await queryAndLog(qNum, client, cardsText, values);
+        const deckResponse = await queryAndLog(qNum, client, decksText, values);
+        res.status(200).json(deckResponse);
+    } catch (err) {
+        console.error(err);
+    }
+});
```

Comment:
Are you also deleting cards from your junction table too?

Reply:
I'm afraid createTables.sql is severely outdated. We don't have a junction table anymore. Sorry!